### PR TITLE
add doc for @@tidb_enable_noop_variables

### PR DESCRIPTION
**Pull request by tangenta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/35496

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -865,6 +865,17 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
 > 该变量只有在默认值 `OFF` 时，才算是安全的。因为设置 `tidb_enable_noop_functions=1` 后，TiDB 会自动忽略某些语法而不报错，这可能会导致应用程序出现异常行为。例如，允许使用语法 `START TRANSACTION READ ONLY` 时，事务仍会处于读写模式。
 
+### `tidb_enable_noop_variables` <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 若该变量值为 `ON`，则 TiDB 和旧版本的 TiDB 行为一致；若该变量值为 `OFF`，TiDB 具有以下行为：
+    * 使用 `SET` 设置 `noop` 的系统变量时会报 "setting *variable_name* has no effect in TiDB" 的 warning。
+    * `SHOW [GLOBAL] VARIABLES` 不再显示 `noop` 的系统变量。
```
Comment (djshow832) [inline system-variables.md:875]:
```suggestion
    * `SHOW [SESSION | GLOBAL] VARIABLES` 不再显示 `noop` 的系统变量。
```

File: system-variables.md [hunk 2]
```diff
@@ -865,6 +865,17 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 >
 > 该变量只有在默认值 `OFF` 时，才算是安全的。因为设置 `tidb_enable_noop_functions=1` 后，TiDB 会自动忽略某些语法而不报错，这可能会导致应用程序出现异常行为。例如，允许使用语法 `START TRANSACTION READ ONLY` 时，事务仍会处于读写模式。
 
+### `tidb_enable_noop_variables` <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 若该变量值为 `ON`，则 TiDB 和旧版本的 TiDB 行为一致；若该变量值为 `OFF`，TiDB 具有以下行为：
```
Comment (djshow832) [inline system-variables.md:873]: It's not good to mention "旧版本" because future users don't know how "旧版本" worked.

File: system-variables.md [hunk 3]
```diff
@@ -865,6 +865,17 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+    * 使用 `SET` 设置 `noop` 的系统变量时会报 "setting *variable_name* has no effect in TiDB" 的 warning。
```
Comment (djshow832) [inline system-variables.md:874]: It also needs `tidb_enable_noop_functions` to be enabled.

File: system-variables.md (hunk 2 above)
Comment (TomShawn) [reply system-variables.md:873]: +1

Comment (tangenta) [reply system-variables.md]: Removed.

File: system-variables.md (hunk 3 above)
Comment (tangenta) [reply system-variables.md:874]: This is not true for some variables like `query_cache_type`:

```
mysql> set tidb_enable_noop_functions=false;
Query OK, 0 rows affected (0.01 sec)

mysql> SET query_cache_type=2;
Query OK, 0 rows affected, 1 warning (0.00 sec)
```

As document mentioned, `tidb_enable_noop_functions` only controls `tx_read_only`、`transaction_read_only`、`offline_mode`、`super_read_only`、`read_only` and `sql_auto_is_null`.

File: system-variables.md [hunk 4]
```diff
@@ -865,6 +865,17 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
 > 该变量只有在默认值 `OFF` 时，才算是安全的。因为设置 `tidb_enable_noop_functions=1` 后，TiDB 会自动忽略某些语法而不报错，这可能会导致应用程序出现异常行为。例如，允许使用语法 `START TRANSACTION READ ONLY` 时，事务仍会处于读写模式。
 
+### `tidb_enable_noop_variables` <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 若该变量值为 `OFF`，TiDB 具有以下行为：
+    * 使用 `SET` 设置 `noop` 的系统变量时会报 "setting *variable_name* has no effect in TiDB" 的 warning。
+    * `SHOW [SESSION | GLOBAL] VARIABLES` 不再显示 `noop` 的系统变量。
```
Comment (djshow832) [inline system-variables.md:875]:
```suggestion
    * `SHOW [SESSION | GLOBAL] VARIABLES` 不显示 `noop` 的系统变量。
```

**Review by djshow832 — COMMENTED** [review]
Rest LGTM

**Review by djshow832 — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -865,6 +865,17 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+### `tidb_enable_noop_variables` <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 若该变量值为 `OFF`，TiDB 具有以下行为：
+    * 使用 `SET` 设置 `noop` 的系统变量时会报 "setting *variable_name* has no effect in TiDB" 的 warning。
+    * `SHOW [SESSION | GLOBAL] VARIABLES` 不显示 `noop` 的系统变量。
+    * 使用 `SELECT` 读取 `noop` 的系统变量时会报 "variable *variable_name* has no effect in TiDB" 的 warning。
+- 用户可以通过 `SELECT * FROM INFORMATION_SCHEMA.CLIENT_ERRORS_SUMMARY_GLOBAL;` 语句来检查 TiDB 实例是否曾设置和读取 `noop` 系统变量。
```
Comment (TomShawn) [inline system-variables.md:877]:
```suggestion
- 若该变量值为 `OFF`，TiDB 具有以下行为：
    * 使用 `SET` 设置 `noop` 的系统变量时会报 `"setting *variable_name* has no effect in TiDB"` 的警告。
    * `SHOW [SESSION | GLOBAL] VARIABLES` 的结果不显示 `noop` 的系统变量。
    * 使用 `SELECT` 读取 `noop` 的系统变量时会报 `"variable *variable_name* has no effect in TiDB"` 的警告。
- 你可以通过 `SELECT * FROM INFORMATION_SCHEMA.CLIENT_ERRORS_SUMMARY_GLOBAL;` 语句来检查 TiDB 实例是否曾设置和读取 `noop` 系统变量。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**